packages/ruleset/src/functions/operation-id-naming-convention.js

```diff
@@ -74,51 +82,100 @@ function operationIdNamingConvention(resolvedSpec) {
+//        |                             |               |              |              |
+// put    | /albums                     | N             | 0            | replace      | replace_albums
+//        | /albums/{id}                | Y             | 1            | replace      | replace_album
+//        | /albums/{id}/genre          | N             | 1            | replace, set | replace_album_genre, set_album_genre
+//        | /albums/{id}/producer/{pid} | Y             | 2+           | add          | add_album_producer
+//        |                             |               |              |              |
+// delete | /albums/{id}                | Y             | 1            | delete       | delete_album
+//        | /albums/{id}/genre          | N             | 1            | unset        | unset_album_genre
+//        | /albums/{id}/producer/{pid} | Y             | 2+           | remove       | remove_album_producer
+//
+// Please try to keep this table in sync with the API Handbook guidance and the logic in the function below.
+//
 function operationIdPassedConventionCheck(
   opKey,
   operationId,
   allPathOperations,
-  pathEndsWithParam
+  pathEndsWithParam,
+  numParamRefs
 ) {
-  // Only consider paths for which
-  // - paths that do not end with param have a GET and POST operation
-  // - paths that end with param have a GET, DELETE, POST, PUT or PATCH.
-
   const verbs = [];
 
-  if (!pathEndsWithParam) {
-    // operationId for GET should starts with "list"
-    if (opKey === 'get') {
-      verbs.push('list');
-    }
-    // operationId for POST should starts with "create" or "add"
-    else if (opKey === 'post') {
-      verbs.push('add');
+  switch (opKey) {
+    case 'get':
+      if (!pathEndsWithParam) {
+        verbs.push('list');
+      } else {
+        verbs.push('get');
+      }
+      if (numParamRefs > 1) {
+        verbs.push('check');
+      }
+      break;
+
+    case 'post':
+      // Need this more lenient logic for VPC.
       verbs.push('create');
-    }
-  } else {
-    // operationId for GET should starts with "get"
-    if (opKey === 'get') {
-      verbs.push('get');
-    }
-    // operationId for DELETE should starts with "delete"
-    else if (opKey === 'delete') {
-      verbs.push('delete');
-    }
-    // operationId for PATCH should starts with "update"
-    else if (opKey === 'patch') {
-      verbs.push('update');
-    }
-    // If PATCH operation doesn't exist for path, POST operationId should start with "update"
-    else if (opKey === 'post') {
-      if (!allPathOperations.includes('patch')) {
+      if (numParamRefs > 0 && !allPathOperations.includes('patch')) {
         verbs.push('update');
       }
-    }
-    // operationId for PUT should starts with "replace"
-    else if (opKey === 'put') {
-      verbs.push('replace');
-    }
+
+      // A normal POST with no param refs should use 'create'.
```

Comment:
The diff makes this a bit confusing, but this commented out block was my initial code for handling a POST operation, but I modified the logic to avoid new warnings in the VPC api.
We need to discuss this...